esx_service/vmdk_ops.py

```diff
@@ -201,22 +201,21 @@ def createVMDK(vmdk_path, vm_name, vol_name, opts={}, vm_uuid=None, tenant_uuid=
 
     # Handle vsan policy
     if kv.VSAN_POLICY_NAME in opts:
-        if not vsan_policy.set_policy_by_name(vmdk_path, opts[kv.VSAN_POLICY_NAME]):
-            # Drop the failed option
-            # A warning is being logged in the called functions
-            del opts[kv.VSAN_POLICY_NAME]
+        set_err = set_policy_to_vmdk(vmdk_path=vmdk_path,
+                            opts=opts,
+                            vol_name=vol_name)
+        if set_err:
+            return set_err
```

Comment:
Shouldn't the VMDK be cleaned and better removed in all cases where error is returned vs. keeping the VMDK around.